PurrSoft/PurrSoft.Api/Controllers/VolunteerController.cs

```diff
@@ -0,0 +1,88 @@
+﻿using AlbumStore.Application.Commands.VolunteerCommands;
+using AlbumStore.Application.Queries.VolunteerQueries;
+using AlbumStore.Application.QueryOverviews;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using PurrSoft.Api.Controllers.Base;
+using PurrSoft.Application.Common;
+using System.Net;
+
+namespace PurrSoft.Api.Controllers;
+
+[Route("api/[controller]")]
+[ApiController]
+public class VolunteerController : BaseController
+{
+    [HttpGet("")]
+    [Authorize(AuthenticationSchemes = "Bearer")]
+    [ProducesResponseType(typeof(List<VolunteerOverview>), (int)HttpStatusCode.OK)]
+    public async Task<CollectionResponse<VolunteerOverview>> GetVolunteers([FromQuery] GetFilteredVolunteersQueries query)
+    {
+        return await Mediator.Send(query, new CancellationToken());
+    }
+
+    [HttpGet("{id}")]
+    [Authorize(AuthenticationSchemes = "Bearer")]
+    [ProducesResponseType(typeof(VolunteerOverview), (int)HttpStatusCode.OK)]
+    [ProducesResponseType((int)HttpStatusCode.NotFound)]
+    public async Task<IActionResult> GetVolunteer(string id)
+    {
+        VolunteerOverview volunteerOverview = await Mediator.Send(new GetVolunteerQuery() { Id = id }, new CancellationToken());
```

Comment:
You can also put in body the get Volunteer. Idk you can proceed here as you would like.